tools/PI/DevHome.PI/Services/WinLogsService.cs

```diff
@@ -1,269 +1,293 @@
+    private Process? _targetProcess;
+    private ETWHelper? _etwHelper;
+    private DebugMonitor? _debugMonitor;
+    private EventViewerHelper? _eventViewerHelper;
+
+    private Thread? _etwThread;
+    private Thread? _debugMonitorThread;
+    private Thread? _eventViewerThread;
+
+    [ObservableProperty]
+    private ObservableCollection<WinLogsEntry> _winLogEntries;
+
+    public WinLogsService()
+    {
+        _winLogEntries = [];
+        _werHelper = Application.Current.GetService<WERHelper>();
+    }
+
+    public void Start(Process process, bool isEtwEnabled, bool isDebugOutputEnabled, bool isEventViewerEnabled, bool isWEREnabled)
+    {
+        _targetProcess = process;
+        Debug.Assert(_targetProcess is not null, "Target Process cannot be null while starting logs");
+
+        if (isEtwEnabled)
+        {
+            StartETWLogsThread();
+        }
+
+        if (isDebugOutputEnabled)
+        {
+            StartDebugOutputsThread();
+        }
+
+        if (isEventViewerEnabled)
+        {
+            StartEventViewerThread();
+        }
+
+        if (isWEREnabled)
+        {
+            StartWER();
+        }
+    }
+
+    public void Stop()
+    {
+        // Stop ETW logs
+        StopETWLogsThread();
+
+        // Stop Debug Outputs
+        StopDebugOutputsThread();
+
+        // Stop Event Viewer
+        StopEventViewerThread();
+
+        // Stop WER
+        StopWER();
+    }
+
+    public void Clear()
+    {
+        WinLogEntries.Clear();
+    }
+
+    public void AddWinLogsEntry(WinLogsEntry entry)
+    {
+        WinLogEntries.Add(entry);
+    }
+
+    public void Dispose()
+    {
+        _etwHelper?.Dispose();
+        _debugMonitor?.Dispose();
+        _eventViewerHelper?.Dispose();
+        GC.SuppressFinalize(this);
+    }
+
+    private void StartETWLogsThread()
+    {
+        // Stop and close existing thread if any
+        StopETWLogsThread();
+
+        // Start a new thread
+        _etwThread = new Thread(() =>
+        {
+            Debug.Assert(_targetProcess is not null, "Target Process cannot be null while starting ETW logs");
+            _etwHelper?.Dispose();
+            _etwHelper = new ETWHelper(_targetProcess);
+            _etwHelper.Start();
+        });
+        _etwThread.Name = EtwLogsName + " Thread";
+        _etwThread.Start();
+    }
+
+    private void StopETWLogsThread()
+    {
+        _etwHelper?.Stop();
+
+        if (Thread.CurrentThread != _etwThread)
+        {
+            _etwThread?.Join();
+        }
+    }
+
+    private void StartDebugOutputsThread()
+    {
+        // Stop and close existing thread if any
+        StopDebugOutputsThread();
+
+        // Start a new thread
+        _debugMonitorThread = new Thread(() =>
+        {
+            Debug.Assert(_targetProcess is not null, "Target Process cannot be null while starting DebugMonitor logs");
+            _debugMonitor?.Dispose();
+            _debugMonitor = new DebugMonitor(_targetProcess);
```

Comment:
Since we can only have one DebugMonitor in the system, it feels like this would be good to treat as a singleton, and we could have multiple subscribers to it (much like the WER service).